### PR TITLE
Refactor Dockerfiles

### PR DESCRIPTION
**Pull request by Gauvino**
Rework Dockerfiles

- Re-organized dockerfile for better syntax
- Changed Debian from 11 (Bullseye) to 12 (Bookworm)
- Changed jellyfin-ffmpeg5 to 6
- Upgrade version of Intel drivers
- Remove wget (only using curl now)

Can be closed https://github.com/jellyfin/jellyfin-metapackages/pull/38

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]

File: Dockerfile.amd64 [hunk 1]
```diff
@@ -3,71 +3,66 @@ ARG TARGET_RELEASE=stable
...
 
 # Install dependencies:
-# curl: healthcheck
 RUN apt-get update \
- && apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg curl wget apt-transport-https \
+ && apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg curl \
  && curl -fsSL https://repo.jellyfin.org/jellyfin_team.gpg.key | gpg --dearmor -o /etc/apt/trusted.gpg.d/debian-jellyfin.gpg \
- && echo 'deb [arch=amd64] https://repo.jellyfin.org/debian bullseye main' > /etc/apt/sources.list.d/jellyfin.list \
- && cat /etc/apt/sources.list.d/jellyfin.list \
+ && echo "deb [arch=$( dpkg --print-architecture )] https://repo.jellyfin.org/debian$( awk -F'=' '/^ID=/{ print $NF }' /etc/os-release ) $( awk -F'=' '/^VERSION_CODENAME=/{ print $NF }' /etc/os-release ) main" > /etc/apt/sources.list.d/jellyfin.list \
```
Comment (Shadowghost) [inline Dockerfile.amd64:29]: We might want to align this with the new apt configs also used in our docs:
```
export VERSION_OS="$( awk -F'=' '/^ID=/{ print $NF }' /etc/os-release )"
export VERSION_CODENAME="$( awk -F'=' '/^VERSION_CODENAME=/{ print $NF }' /etc/os-release )"
export DPKG_ARCHITECTURE="$( dpkg --print-architecture )"
cat <<EOF | sudo tee /etc/apt/sources.list.d/jellyfin.sources
Types: deb
URIs: https://repo.jellyfin.org/${VERSION_OS}
Suites: ${VERSION_CODENAME}
Components: main
Architectures: ${DPKG_ARCHITECTURE}
Signed-By: /etc/apt/keyrings/jellyfin.gpg
EOF
```

File: Dockerfile.amd64 [hunk 2]
```diff
@@ -3,71 +3,66 @@ ARG TARGET_RELEASE=stable
...
+
 # https://github.com/intel/compute-runtime/releases
-ARG GMMLIB_VERSION=22.3.0
-ARG IGC_VERSION=1.0.14828.8
-ARG NEO_VERSION=23.30.26918.9
-ARG LEVEL_ZERO_VERSION=1.3.26918.9
+ARG GMMLIB_VERSION=22.3.11.ci17757293
+ARG IGC_VERSION=1.0.15136.22
+ARG NEO_VERSION=23.39.27427.23
+ARG LEVEL_ZERO_VERSION=1.3.27427.23
```
Comment (Shadowghost) [inline Dockerfile.amd64:23]: There was another update (see #38)

File: Dockerfile.arm64 [hunk 3]
```diff
@@ -4,48 +4,45 @@ ARG TARGET_RELEASE=stable
...
+ENV JELLYFIN_CACHE_DIR=/cache
 
 # Install dependencies:
-#   curl: healcheck
 RUN apt-get update \
- && apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg curl wget \
+ && apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg curl \
  && curl -fsSL https://repo.jellyfin.org/jellyfin_team.gpg.key | gpg --dearmor -o /etc/apt/trusted.gpg.d/debian-jellyfin.gpg \
- && echo 'deb [arch=arm64] https://repo.jellyfin.org/debian bullseye main' > /etc/apt/sources.list.d/jellyfin.list \
+ && echo "deb [arch=$( dpkg --print-architecture )] https://repo.jellyfin.org/debian$( awk -F'=' '/^ID=/{ print $NF }' /etc/os-release ) $( awk -F'=' '/^VERSION_CODENAME=/{ print $NF }' /etc/os-release ) main" > /etc/apt/sources.list.d/jellyfin.list \
```
Comment (Shadowghost) [inline Dockerfile.arm64:22]: Same as above

File: Dockerfile.armhf [hunk 4]
```diff
@@ -4,48 +4,45 @@ ARG TARGET_RELEASE=stable
...
+ENV JELLYFIN_CACHE_DIR=/cache
 
 # Install dependencies:
-#   curl: healthcheck
 RUN apt-get update \
- && apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg curl wget \
+ && apt-get install --no-install-recommends --no-install-suggests -y ca-certificates gnupg curl \
  && curl -fsSL https://repo.jellyfin.org/jellyfin_team.gpg.key | gpg --dearmor -o /etc/apt/trusted.gpg.d/debian-jellyfin.gpg \
- && echo 'deb [arch=armhf] https://repo.jellyfin.org/debian bullseye main' > /etc/apt/sources.list.d/jellyfin.list \
+ && echo "deb [arch=$( dpkg --print-architecture )] https://repo.jellyfin.org/debian$( awk -F'=' '/^ID=/{ print $NF }' /etc/os-release ) $( awk -F'=' '/^VERSION_CODENAME=/{ print $NF }' /etc/os-release ) main" > /etc/apt/sources.list.d/jellyfin.list \
```
Comment (Shadowghost) [inline Dockerfile.armhf:22]: Same as above

**Review by Shadowghost — COMMENTED** [review]
Additionally there was a new dependency added for rkmpp which is missing here too.

File: Dockerfile.amd64 (hunk 1 above)
Comment (Gauvino) [reply Dockerfile.amd64:29]: Which documentation please

Comment (Gauvino) [reply Dockerfile.amd64:29]: Never mind

**Comment by Gauvino** [comment]
Ready to merge !

image

**Review by Shadowghost — APPROVED** [review]

**Comment by Gauvino** [comment]
> Sorry to interrupt but can i help with testing? I need this merge to be done cause of Issue [#10674 AMD VAAPI transcoding](https://github.com/jellyfin/jellyfin/issues/10674). What do i need to do? I know a bit about docker but i didnt understand wich files are needed within the dockerfile to build the new image correctly so i can test it - especially the vulkan api catched my eye ^^ Thx and thx for your work on jelly <3

This PR is ready to merge I don't have to make any other change live checked everything 5 times we now have to wait for the team review and merge and if feel free to ask me some questions !

**Comment by Gauvino** [comment]
I've checked and it need update on intel driver I'm coming back to my home in 5h and I make the new change

**Review by gnattu — APPROVED** [review]
LGTM

**Comment by DerP4si** [comment]
> > Sorry to interrupt but can i help with testing? I need this merge to be done cause of Issue [#10674 AMD VAAPI transcoding](https://github.com/jellyfin/jellyfin/issues/10674). What do i need to do? I know a bit about docker but i didnt understand wich files are needed within the dockerfile to build the new image correctly so i can test it - especially the vulkan api catched my eye ^^ Thx and thx for your work on jelly <3
> 
> This PR is ready to merge I don't have to make any other change live checked everything 5 times we now have to wait for the team review and merge and if feel free to ask me some questions !

Gtk... So is it enough to dl your repo and "execute" the right dockerfile to have the newest jellyfin up n running?

**Comment by Gauvino** [comment]
> > > Sorry to interrupt but can i help with testing? I need this merge to be done cause of Issue [#10674 AMD VAAPI transcoding](https://github.com/jellyfin/jellyfin/issues/10674). What do i need to do? I know a bit about docker but i didnt understand wich files are needed within the dockerfile to build the new image correctly so i can test it - especially the vulkan api catched my eye ^^ Thx and thx for your work on jelly <3
> 
> > 
> 
> > This PR is ready to merge I don't have to make any other change live checked everything 5 times we now have to wait for the team review and merge and if feel free to ask me some questions !
> 
> 
> 
> Gtk... So is it enough to dl your repo and "execute" the right dockerfile to have the newest jellyfin up n running?

Of course no problem I've tested the image and I didn't have any problem, say to me if it fix your problem

**Pull request closed without merging**